### PR TITLE
Remake game button

### PR DESCRIPTION
**Pull request by 5720axolotls**
We update the GameManager to accept a list of initialPlayers so that you can hit a button to remake the game and it will start populated. I think this is better than immediately starting the game with the same players since people might join/leave.

As for getting this list of players passed along, I think it's not unreasonable for it to be encoded as a list in the url that hits the GameHome page (which calls GameManager).

Screenshot 2024-02-28 at 11 39 35 PM

Verification
- [x] needs to be tested
- [x] test that it works in prod by merging

### CODE REVIEW

File: app/game/[gameId]/GameManager.tsx [hunk 1]
```diff
@@ -2,8 +2,8 @@
 import React from 'react'
 import { createGameCode, startGame } from '../index'
 
-export const GameManager = (props: { gameId: string }) => {
-  const [players, setPlayers] = React.useState<string[]>([]);
+export const GameManager = (props: { gameId: string, initialPlayers: string[] }) => {
```
Comment (cailyncodes) [inline app/game/[gameId]/GameManager.tsx]: Nit: Can we de-structure the object within the input, so something like: 
```
({ gameId, initialPlayers }: { gameId: string, initialPlayers: string[] })
```
And potentially lift the type to an alias like:
```
interface GameManagerProps {
 gameId: string;
 initialPlayers: string[];
}
```

File: app/game/[gameId]/GameManager.tsx [hunk 2]
```diff
@@ -18,6 +18,7 @@ export const GameManager = (props: { gameId: string }) => {
   }
 
   const startNewGame = async (_: FormData) => {
+    console.log("calling startNewGame with players: ", players)
```
Comment (cailyncodes) [inline app/game/[gameId]/GameManager.tsx]: Nit: remove

File: app/game/[gameId]/view/page.tsx [hunk 3]
```diff
@@ -1,18 +1,40 @@
-import { getGame } from '../../index'
+"use client"
```
Comment (cailyncodes) [inline app/game/[gameId]/view/page.tsx]: Nit: I would rather move some of this client side logic to a new component and use that within this server side component, as opposed to making the entire thing client

File: app/game/index.ts [hunk 4]
```diff
@@ -49,9 +63,9 @@ export async function startGame(data: { gameId: string, players: string[] }) {
   })
 }
 
-export async function getGame(gameId: string) {
+export async function getGame(gameId: string): Promise<Game> {
   const response = await fetch(
-    "https://kslx3eprjeoij69w.public.blob.vercel-storage.com/" + gameId + ".json",
+    "https://spwamd4ap0dqqd0y.public.blob.vercel-storage.com/" + gameId + ".json",
```
Comment (cailyncodes) [inline app/game/index.ts]: Nit: Make sure to update before merging

File: app/game/index.ts [hunk 5]
```diff
@@ -72,7 +86,7 @@ export async function createGameCode(gameId: string) {
 }
 
 export async function getGameId(gameCode: string) {
-  const gameId = await kv.get(gameCode)
+  const gameId = await kv.get(gameCode.toUpperCase())
```
Comment (cailyncodes) [inline app/game/index.ts]: Let's merge in master to avoid this change in the diff

File: app/game/[gameId]/GameManager.tsx (hunk 1 above)
Comment (5720axolotls) [reply app/game/[gameId]/GameManager.tsx]: I agree honestly. I think the props probably should have been lifted in the first place... 👀

File: app/game/[gameId]/view/page.tsx [hunk 6]
```diff
@@ -1,18 +1,40 @@
...
 }: {
   params: { gameId: string },
   searchParams: { [key: string]: string | string[] | undefined }
 }) {
-  const game = await getGame(params.gameId)
+
+  const [game, setGame] = React.useState<Game>()
   const gameCode = searchParams['gameCode']
 
+  // make useEffect to get the game data
```
Comment (dblackey) [inline app/game/[gameId]/view/page.tsx]: Capitalize comment

File: app/game/[gameId]/view/page.tsx [hunk 7]
```diff
@@ -1,18 +1,40 @@
...
+  React.useEffect(() => {
+    const fetchGame = async () => {
+      const gameData = await getGame(params.gameId)
+      setGame(gameData)
+    }
+    fetchGame()
+  }, [])
```
Comment (dblackey) [inline app/game/[gameId]/view/page.tsx]: semicolon

**Review by dblackey — CHANGES_REQUESTED** [review]
Please leave more thorough verification steps.

File: app/game/[gameId]/view/page.tsx (hunk 6 above)
Comment (elenaswecker) [reply app/game/[gameId]/view/page.tsx]: lowercase comment

File: app/game/[gameId]/view/page.tsx (hunk 3 above)
Comment (5720axolotls) [reply app/game/[gameId]/view/page.tsx]: :0 ok, not entirely sure how to do this one, but I'll give it a shot!

**Review by cailyncodes — APPROVED** [review]
Looks great! Thanks for doing this!

**Pull request merged**